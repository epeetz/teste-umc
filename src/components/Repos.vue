<template>
	<div class="container mb-3">
		<div class="d-flex mb-3">
			<div class="mr-auto">
				<h3>Repositories by {{ reformattedSearchString }}</h3>
			</div>
		</div>
		<div class="card-columns">
			<div class="card" v-bind:key="repo.id" v-for="repo in repos">
				<div>
					<div class="card-body">
						<h5 class="card-title">
							<a
							:href="repo.html_url"
							class="card-link"
							target="_blank">{{ repo.name }}</a>
						</h5>
						<h6 class="card-subtitle mb-2 text-muted" >{{ repo.created_at | formatDate }}</h6>
						<p class="card-text">{{ repo.description }}</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</template>

<script>

export default {
	name: 'Repos',
	props: ['repos','reformattedSearchString']
};
</script>

<style scoped>
button:focus {
	box-shadow: none !important;
}
</style>