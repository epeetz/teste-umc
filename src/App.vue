<template>
	<div id="app">
		<Header/>
		<SearchForm v-on:search="search"/>
		<Repos
		v-if="repos.length > 0"
		v-bind:repos="repos"
		v-bind:reformattedSearchString="reformattedSearchString" />
	</div>
</template>

<script>
import Header from './components/Header.vue';
import SearchForm from './components/SearchForm.vue';
import Repos from './components/Repos.vue';
import axios from 'axios';

export default {
	name: 'app',
	components: {
		Header,
		SearchForm,
		Repos
	},
	data() {
		return {
			repos: [],
			reformattedSearchString: '',
			api: {
				baseUrl: 'https://api.github.com/users',
				q: 'epeetz',
				type: 'repos'
			}
		};
	},
	mounted() {
		this.getData('https://api.github.com/users/epeetz/repos');
		this.reformattedSearchString = '[ "epeetz" ]';
	},
	methods: {
		search(searchParams) {
			this.reformattedSearchString = searchParams;
			this.api.q = searchParams;
			const { baseUrl, q, type } = this.api;
			const apiUrl = `${baseUrl}/${q}/${type}`;
			this.getData(apiUrl);
		},
		getData(apiUrl) {
			axios
			.get(apiUrl)
			.then((response) => {
				this.repos = response.data;
			})
			.catch(error => console.log(error));
		}
	}
}
</script>

<style>
#app {
	font-family: 'Avenir', Helvetica, Arial, sans-serif;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
	text-align: center;
	color: #2c3e50;
}
</style>
