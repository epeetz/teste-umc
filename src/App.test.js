import { shallowMount } from '@vue/test-utils';
import App from './App';
import Header from './components/Header';
import SearchForm from './components/SearchForm';
import Repos from './components/Repos';

let wrapper;

beforeEach(() => {
    wrapper = shallowMount(App, {
        data: () => ({
            api: {
                q: 'epeetz'
            }
        }),
        propsData: {},
        mocks: {},
        stubs: {},
        methods: {},
    });
});

afterEach(() => {
    wrapper.destroy();
});

describe('App', () => {
    test('is a Vue instance', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });
    test('rendering', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
    test('renders main child components', () => {
        const appHeader = wrapper.find(Header);
        const appSearchForm = wrapper.find(SearchForm);
        expect(appHeader.exists()).toBe(true);
        expect(appSearchForm.exists()).toBe(true);
    });
});