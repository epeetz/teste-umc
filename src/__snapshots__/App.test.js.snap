// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering 1`] = `"<div id=\\"app\\"><header-stub></header-stub> <searchform-stub></searchform-stub> <!----></div>"`;
